.github/workflows/check-lib.yml

```diff
@@ -0,0 +1,272 @@
+
+  rustfmt:
+    name: Format
+    needs: [MSRV]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Run cargo fmt
+        run: cargo fmt --all -- --check
+
+  min-vers:
+    name: Minimal crate versions
+    needs: [MSRV]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          override: true
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Install cargo-hack
+        uses: actions-rs/install@v0.1
+        with:
+          crate: cargo-hack
+
+      - name: Check minimal versions
+        run: |
+          # Ignore dev-deps
+          cargo hack --remove-dev-deps --workspace
+          # Ignore examples
+          sed -i '/examples/d' Cargo.toml
+          # Update Cargo.lock to minimal version dependencies.
+          cargo update -Z minimal-versions
+          cargo check --all-features
+
+  # Tests
+  test:
+    name: Test
+    needs: [MSRV]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Run cargo test
+        run: cargo test
+
+  test-features:
+    name: Test ${{ matrix.package }} feat. ${{ matrix.features }}
+    needs: [test]
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        package:
+          - http
+          - gateway
+          - lavalink
+        features:
+          - native
+          - rustls
+          - simd-json
+
+        include:
+          - package: http
+            features: simd-json
+            additional: --features rustls
+          - package: gateway
+            features: rustls
+            additional: --features zlib-stock
+          - package: gateway
+            features: native
+            additional: --features zlib-stock
+          - package: gateway
+            features: simd-json
+            additional: --features rustls,zlib-stock
+          - package: lavalink
+            additional: --features http-support
+
+          - package: util
+            features: full
+
+        exclude:
+          - package: lavalink
+            features: simd-json
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Test ${{ matrix.package }} feat. ${{ matrix.features }}
+        working-directory: ${{ matrix.package }}
+        run: |
+          cargo test --no-default-features --features ${{ matrix.features }} ${{ matrix.additional }}
+
+  coverage:
+    name: Coverage
+    runs-on: ubuntu-latest
+    # Don't collect coverage if tests failed
+    needs: [test]
```

Comment:
should this also include MSRV just for completions sake?

Reply:
This ends up looking fine to me:
<img width="1018" alt="image" src="https://user-images.githubusercontent.com/1760581/142677308-c0068906-a5ee-4a47-b310-659ba8b17648.png">